src/LiveComponent/src/Form/Type/MultiStepType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefault('current_step_name', static function (Options $options): string {
+                return array_key_first($options['steps']);
+            })
+            ->setRequired('steps');
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $options['steps'][$options['current_step_name']]($builder);
```

Comment:
```suggestion
        $options['steps'][$options['current_step_name']]($builder, $options);
```

IMO, passing `$options` to keep consistent with `buildForm` method can be good.

Reply:
sure we can do that but my question is. Why would it make sense to pass the options of the "parent" form StepType to the children? you got an example or use case where it might be useful?

Reply:
It is more to be consistent with how work the `buildForm` method, where you have access to options.

In your PR, you cannot set your form options yet, so there is no direct example. `instantiateForm` method pass an empty `$options` (except the `current_step_name`) to `FormFactory::create`.

You may want to create step form that is configurable, either from:
- your live component (not yet possible)
- using `FormTypeExtension` from Form component

Is it clearer that way?